### PR TITLE
Add full export history page (no pagination)

### PR DESCRIPTION
**Pull request by debitan**
## Description of change

This adds the full export history page. In this first iteration there is no pagination or axios calls. It simply fetches all the data and displays it in one go. Pagination will be in a later PR. 

This is also a hidden URL at the moment. I will create a seperate PR to add the link to this page to the export page. 

All additions and deletions are currently shown as individual events. There is a ticket for the backend to group these together to reflect when countries were added and deleted in a single user event. I will further iterate this front end feature to incorporate this when it is ready. 

## Test instructions

You should see the export countries history using the collectionList component when going to /companies/:companyId/exports/history

There is only history from when we merged the history PR last week - so you may have to add and delete some export countries to see the feature in action. 

## Screenshots

FireShot Capture 034 - Full export history - Exports - One List Subsidiary Ltd - Companies -_ - localhost

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [ ] Has the branch been rebased to master?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or Acceptance)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

File: src/apps/companies/apps/exports/client/ExportsFullHistory.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+
+import { H3 } from '@govuk-react/heading'
+
+import { CollectionList } from 'data-hub-components'
+
+const Wrapper = styled('div')`
+  margin-top: 16px;
```
Comment (ian-leggett) [inline src/apps/companies/apps/exports/client/ExportsFullHistory.jsx]: We should use spacing units here - `import { SPACING } from '@govuk-react/constants'`

File: src/apps/companies/apps/exports/client/ExportsFullHistory.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+`
+
+function ExportFullHistory({ data }) {
+  return (
+    <Wrapper>
+      <H3>Export markets history</H3>
```
Comment (ian-leggett) [inline src/apps/companies/apps/exports/client/ExportsFullHistory.jsx]: `<h2>` as mentioned above

File: src/apps/companies/apps/exports/client/ExportsFullHistory.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      <CollectionList
+        items={data.results}
```
Comment (ian-leggett) [inline src/apps/companies/apps/exports/client/ExportsFullHistory.jsx]: I think `data.results` should be populated via a saga and injected into the redux store.

File: src/apps/companies/apps/exports/controller.js [hunk 4]
```diff
@@ -81,6 +85,22 @@ function renderExports(req, res) {
     })
 }
 
+async function renderFullExportHistory(req, res) {
+  const { token } = req.session
+  const { company } = res.locals
+  const data = await getFullExportHistory(token, company.id)
```
Comment (ian-leggett) [inline src/apps/companies/apps/exports/controller.js]: This should be a Redux saga

File: src/apps/companies/apps/exports/client/ExportsFullHistory.jsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+
+import { H3 } from '@govuk-react/heading'
```
[the remark this answers is not in the document]
Comment (paulgain) [reply src/apps/companies/apps/exports/client/ExportsFullHistory.jsx]: All `node_module` imports should be grouped together.

File: src/apps/companies/apps/exports/repos.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+const config = require('../../../../config')
+const { authorisedRequest } = require('../../../../lib/authorised-request')
+
+module.exports = {
+  getFullExportHistory: (token, companyId) => {
```
Comment (ian-leggett) [inline src/apps/companies/apps/exports/repos.js]: We don't actually need this now, this should be a Redux saga which populates state in the Redux store

File: src/apps/companies/apps/exports/repos.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+    return authorisedRequest(token, {
+      url: `${config.apiRoot}/v4/search/export-country-history`,
+      method: 'POST',
+      body: {
+        company: `${companyId}`,
```
Comment (paulgain) [inline src/apps/companies/apps/exports/repos.js]: `companyId ` will suffice.

File: src/apps/companies/apps/exports/repos.js (hunk 6 above)
Comment (debitan) [reply src/apps/companies/apps/exports/repos.js]: even for this iteration that doesn't have any state? is this maybe a consideration for the next PR - https://github.com/uktrade/data-hub-frontend/pull/2428

File: src/apps/companies/apps/exports/transformer.js [hunk 8]
```diff
@@ -25,51 +36,78 @@ function getCountriesFields(company) {
...
-      company.export_experience_category &&
-      company.export_experience_category.name,
-  }
+  return [country, historyTypeText, typeText].join(' ')
+}
 
-  const greatProfile = {
-    name: exportDetailsLabels.greatProfile,
-    value: company.great_profile_status,
+function createCountry(item) {
```
Comment (paulgain) [inline src/apps/companies/apps/exports/transformer.js]: You could destructure `item` here.

File: src/apps/companies/apps/exports/transformer.js [hunk 9]
```diff
@@ -25,51 +36,78 @@ function getCountriesFields(company) {
...
+}
 
-  const exportPotential = {
-    name: exportDetailsLabels.exportPotential,
-    value:
-      exportPotentialLabels[company.export_potential] &&
-      exportPotentialLabels[company.export_potential].text,
-  }
+module.exports = {
+  transformFullExportHistory: (data) => ({
```
Comment (paulgain) [inline src/apps/companies/apps/exports/transformer.js]: You could destructure `data` here.

File: src/apps/companies/apps/exports/views/full-history.njk [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+{% block body_main_content %}
+
+  <div class="govuk-grid-column-full">
+            {% component 'react-slot', {
+              id: 'company-export-full-history',
+              props: props
+            }
+            %}
+  </div>
```
Comment (paulgain) [inline src/apps/companies/apps/exports/views/full-history.njk]: The content of this `div` needs aligning correctly.

File: test/functional/cypress/specs/companies/export-spec.js [hunk 11]
```diff
@@ -202,8 +203,96 @@ describe('Companies Export', () => {
...
     })
   })
 })
+
+describe('Companies Export Countries', () => {
+  const countrySelectors = exportSelectors.countries
+
+  context('when there is no history', () => {
+    before(() => {
+      cy.visit(`/companies/${fixtures.company.lambdaPlc.id}/exports/history`)
```
Comment (paulgain) [inline test/functional/cypress/specs/companies/export-spec.js]: urls module

File: src/apps/companies/apps/exports/repos.js (hunk 6 above)
Comment (ian-leggett) [reply src/apps/companies/apps/exports/repos.js]: Fair enough lets tackle the Redux bits in the second PR

File: test/functional/cypress/specs/companies/export-spec.js [hunk 12]
```diff
@@ -202,8 +203,96 @@ describe('Companies Export', () => {
...
+
+    it('renders the collection list with the 7 results', () => {
+      cy.contains('7 results').should('exist')
+      cy.get(countrySelectors.listItemHeadings).should('have.length', 7)
+
+      cy.contains('Argentina added to currently exporting').should('exist')
+      cy.contains('Argentina added to currently exporting')
+        .siblings()
+        .should('contain', 'By DIT Staff')
+        .should('contain', 'Date 6 Feb 2020, 4:06pm')
```
Comment (paulgain) [inline test/functional/cypress/specs/companies/export-spec.js]: L255 and L256 do the same thing, you can remove L255 as it duplicating the next line.

`Contains`: 
Get the DOM element containing the text
https://docs.cypress.io/api/commands/contains.html#Syntax

`should`:
Asserts that the target is neither null nor undefined (Chai: https://www.chaijs.com/api/assert/#method_exists)

The same applies to everything below.

File: test/functional/cypress/specs/companies/export-spec.js [hunk 13]
```diff
@@ -202,8 +203,96 @@ describe('Companies Export', () => {
...
+        .should('contain', 'By DIT Staff')
+        .should('contain', 'Date 6 Feb 2020, 3:40pm')
+      cy.contains('Albania added to currently exporting').should('exist')
+      cy.contains('Albania added to currently exporting')
+        .siblings()
+        .should('contain', 'By DIT Staff')
+        .should('contain', 'Date 6 Feb 2020, 3:40pm')
+    })
+  })
+})
```
Comment (paulgain) [inline test/functional/cypress/specs/companies/export-spec.js]: I think we're over using contains here. The problem with contains is that it queries the entire document which is not performant (I don't have any stats to back this up). Personally, we are better off using contains sparingly, we could use it to initially to find a DOM element (even better, be specific with `get`) and then cycle though by chaining all the commands/asserts like @peterhudec does. This also ensures all DOM elements are in the correct order.
https://github.com/uktrade/data-hub-frontend/blob/master/test/functional/cypress/specs/companies/dnb-hierarchy-spec.js#L49-L87

File: src/apps/companies/apps/exports/client/ExportsFullHistory.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (debitan) [reply src/apps/companies/apps/exports/client/ExportsFullHistory.jsx]: group the imports. 

H3 is how it is in our design and it is also consistent with the headers within the tab navigation.

File: src/apps/companies/apps/exports/client/ExportsFullHistory.jsx (hunk 1 above)
Comment (debitan) [reply src/apps/companies/apps/exports/client/ExportsFullHistory.jsx]: I'll use SCALE_3 which is 15px and so close enough to the original 16px I chose.

File: src/apps/companies/apps/exports/client/ExportsFullHistory.jsx (hunk 2 above)
Comment (debitan) [reply src/apps/companies/apps/exports/client/ExportsFullHistory.jsx]: see comment above

File: test/functional/cypress/specs/companies/export-spec.js (hunk 13 above)
Comment (debitan) [reply test/functional/cypress/specs/companies/export-spec.js]: We can talk about this more away from review, but I think as this is quite a dynamic page I would have to essentially reimplement the transformer function here to be able to cycle programmatically through and assert what should appear. This feels a bit too much like mocking. All I want to check here is that the text is appearing in the way expected for the user and that the outcome of the transformer is correct. The order of the items is dictated by the backend and is tested there. 
As for speed, these tests execute almost instantaneously, so I don't think we have much to worry about there. 
I don't believe this should be a blocker for this PR.

File: src/apps/companies/apps/exports/client/ExportsFullHistory.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (ian-leggett) [reply src/apps/companies/apps/exports/client/ExportsFullHistory.jsx]: The size of the header (look and feel) is fine I was just thinking accessibility. It looks fine but I would use `<H2 size={LEVEL_SIZE[3]}>` Ideally you can't have a H3 following a H1

[the remark this answers is not in the document]
Comment (debitan) [reply src/apps/companies/apps/exports/client/ExportsFullHistory.jsx]: Gotcha! Will fix that.

**Pull request merged**